src/main/java/org/csanchez/jenkins/plugins/kubernetes/pod/retention/Reaper.java

```diff
@@ -139,7 +140,7 @@ public void eventReceived(Watcher.Action action, Pod pod) {
                         continue;
                     }
                     KubernetesSlave ks = (KubernetesSlave) n;
-                    if (ks.getNamespace().equals(ns) && ks.getPodName().equals(name)) {
+                    if (StringUtils.equals(ks.getNamespace(), ns) && StringUtils.equals(ks.getPodName(), name)) {
```

Comment:
Under which circumstances would either of these be null?

You can use `java.util.Objects.equals` BTW.

Reply:
I think there can be a race condition between Reaper and KubernetesLauncher. In particular, namespace is not set immediately when KubernetesSlave is created. And i've definitely seen occurrences of the NPE when launching a lot of agents concurrently.